lib/mongo/error/read_write_retryable.rb

```diff
@@ -0,0 +1,104 @@
+# frozen_string_literal: true
+# encoding: utf-8
+
+# Copyright (C) 2019-2022 MongoDB Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+module Mongo
+  class Error < StandardError
+
+    # A module encapsulating functionality to indicate whether errors are
+    # retryable.
+    #
+    # @api private
+    module ReadWriteRetryable
+
+      # Error codes and code names that should result in a failing write
+      # being retried.
+      #
+      # @api private
```

Comment:
And here also copy-paste https://github.com/mongodb/mongo-ruby-driver/blob/master/lib/mongo/server/connection_base.rb#L23-L27.